modAionImpl/src/org/aion/zero/impl/blockchain/AionPendingStateImpl.java

```diff
@@ -531,67 +555,79 @@ public synchronized AionBlock getBestBlock() {
                         (limit-- > 0) &&
                         (txBuffer == null ? txPool.size() : txPool.size() + txBuffer.size())
                             < MAX_VALIDATED_PENDING_TXS);
-                } else if (bestRepoNonce(tx.getFrom()).compareTo(txNonce) < 1) {
-                    // repay Tx
-                    if (addPendingTransactionImpl(tx, txNonce)) {
-                        newPending.add(tx);
+                }
+            }
+            // This case happens when this tx was received before, but never sealed,
+            // typically because of low energy
+            else if (bestRepoNonce(txFrom).compareTo(txNonce) < 1) {
+                // repay Tx
+                TxResponse implResponse = addPendingTransactionImpl(tx, txNonce);
+                if (implResponse.equals(TxResponse.SUCCESS)) {
+                    newPending.add(tx);
+                    txResponses.add(TxResponse.REPAID);
```

Comment:
Yeah, this is a little messy. Basically, if the transaction is successfully added, we want to respond with REPAID. However, we cannot expect addPendingTransactionImpl() to return REPAID as its return value, because that function doesn't know we're trying to repay the tx. 

So if addPendingTransactionImpl() returns SUCCESS, then we should return REPAID. In all other cases, we return whatever addPendingTransactionImpl() returned to us.

If you think it's messy, we can put a check in addPendingTransactionImpl() that would return REPAID in this case.